eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py

```diff
@@ -69,81 +99,146 @@ def get_assembly_info_and_date_ranges(all_log_files):
+                'from': job_id_time_ranges['PROCESS_RS_SPLIT_CANDIDATES_STEP'],
+                'to': job_id_time_ranges['CLEAR_RS_MERGE_AND_SPLIT_CANDIDATES_STEP']
+                if "CLEAR_RS_MERGE_AND_SPLIT_CANDIDATES_STEP" in job_id_time_ranges
+                else job_id_time_ranges["last_timestamp"]
             }
         # new_ss_clustered
-        if 'CLUSTERING_NON_CLUSTERED_VARIANTS_FROM_MONGO_STEP' in log_metric_date_range:
-            ranges_per_assembly[assembly_accession]['metrics']['new_ss_clustered'][log_file] = {
-                'from': log_metric_date_range['CLUSTERING_NON_CLUSTERED_VARIANTS_FROM_MONGO_STEP'],
-                'to': log_metric_date_range['CLUSTER_UNCLUSTERED_VARIANTS_JOB']["completed"]
-                if "completed" in log_metric_date_range['CLUSTER_UNCLUSTERED_VARIANTS_JOB']
-                else log_metric_date_range["last_timestamp"]
+        if 'CLUSTERING_NON_CLUSTERED_VARIANTS_FROM_MONGO_STEP' in job_id_time_ranges:
+            ranges_per_assembly[assembly_accession]['metrics']['new_ss_clustered'][job_id] = {
+                'from': job_id_time_ranges['CLUSTERING_NON_CLUSTERED_VARIANTS_FROM_MONGO_STEP'],
+                'to': job_id_time_ranges['CLUSTER_UNCLUSTERED_VARIANTS_JOB']["completed"]
+                if "completed" in job_id_time_ranges['CLUSTER_UNCLUSTERED_VARIANTS_JOB']
+                else job_id_time_ranges["last_timestamp"]
             }
         # new_ss_clustered
-        if study_clustering_step_name in log_metric_date_range:
-            ranges_per_assembly[assembly_accession]['metrics']['new_ss_clustered'][log_file] = {
-                'from': log_metric_date_range[study_clustering_step_name],
-                'to': log_metric_date_range[study_clustering_job_name]["completed"]
-                if "completed" in log_metric_date_range[study_clustering_job_name]
-                else log_metric_date_range["last_timestamp"]
+        if study_clustering_step_name in job_id_time_ranges:
+            ranges_per_assembly[assembly_accession]['metrics']['new_ss_clustered'][job_id] = {
+                'from': job_id_time_ranges[study_clustering_step_name],
+                'to': job_id_time_ranges[study_clustering_job_name]["completed"]
+                if "completed" in job_id_time_ranges[study_clustering_job_name]
+                else job_id_time_ranges["last_timestamp"]
             }
     return ranges_per_assembly
 
 
+def get_time_ranges_by_job_id_from_job_tracker(assembly_accession, private_config_xml_file,
+                                               prev_release_end_timestamp_for_assembly: datetime = None):
+    # Job and step start/end times by job ID
+    time_ranges_by_job_id = defaultdict(dict)
+    url, username, password = get_accession_pg_creds_for_profile("production_processing", private_config_xml_file)
+    accessioning_job_tracker_handle = psycopg2.connect(urlsplit(url).path, user=username, password=password,
```

Comment:
Should put in a `with` statement, or just make sure we close the connection at the end

Reply:
Addressed